src/pretix/base/timeframes.py

```diff
@@ -0,0 +1,388 @@
+    if describe:
+        details = f'{describe(start(now()))}: {details}'
+    return f'{label} ({details})'
+
+
+class DateFrameField(forms.MultiValueField):
+    def __init__(self, *args, **kwargs):
+        include_future_frames = kwargs.pop('include_future_frames')
+
+        top_choices = [('custom', gettext_lazy('Custom timeframe'))]
+        if not kwargs.get('required', True):
+            top_choices.insert(0, ('unset', pgettext_lazy('reporting_timeframe', 'All time')))
+
+        _choices = []
+        for grouper, group in groupby(REPORTING_DATE_TIMEFRAMES, key=lambda i: i[5]):
+            _choices.append((grouper, [
+                (identifier, _describe_timeframe(label, start, end, future, describe))
+                for identifier, label, start, end, future, group, describe in group
+                if include_future_frames or not future
+            ]))
+
+        timeframe_choices = [
+            ('', top_choices)
+        ] + _choices
+
+        fields = (
+            forms.ChoiceField(
+                choices=timeframe_choices,
+                required=True
+            ),
+            forms.DateField(
+                required=False
+            ),
+            forms.DateField(
+                required=False
+            ),
+        )
+        if 'widget' not in kwargs:
+            kwargs['widget'] = DateFrameWidget(timeframe_choices=timeframe_choices)
+        kwargs.pop('max_length', 0)
+        kwargs.pop('empty_value', 0)
+        super().__init__(
+            fields=fields, require_all_fields=False, *args, **kwargs
+        )
+
+    def compress(self, data_list):
+        if not data_list:
+            return None
+        if data_list[0] == 'unset':
+            return None
+        elif data_list[0] == 'custom':
+            return f'{data_list[1].isoformat()}/{data_list[2].isoformat()}'
+        else:
+            return data_list[0]
+
+    def has_changed(self, initial, data):
+        if initial is None:
+            initial = self.widget.decompress(initial)
+        return super().has_changed(initial, data)
+
+    def clean(self, value):
+        if value[0] == 'custom':
+            if not value[1] or not value[2]:
+                raise ValidationError(self.error_messages['incomplete'])
+            # todo validate start<end
+        return super().clean(value)
+
+
+def resolve_timeframe_to_dates_inclusive(ref_dt, frame, timezone) -> Tuple[date, date]:
+    """
+    Given a serialized timeframe, evaluate it relative to `ref_dt` and return a tuple of dates
+    where the first element ist the first possible date value within the timeframe and the second
+    element is the last possible date value in the tiemframe.
+    """
+    if isinstance(ref_dt, datetime):
+        ref_dt = ref_dt.astimezone(timezone).date()
+    if "/" in frame:
+        start, end = frame.split("/", 1)
+        return date.fromisoformat(start), date.fromisoformat(end)
```

Comment:
What about allowing incomplete dateranges? Might be helpful for certain reports. Would need some changes in other parts, but should be doable? I could try and add it, if you want.

```suggestion
        return date.fromisoformat(start) if start else None, date.fromisoformat(end) if end else None
```